### ISSUE TITLE
A11y_Azure Tools Storage Explorer_Disk_Delete Disk_Screen reader: Voice over is not announcing the Name "Microsoft Azure Storage explorer" of the dialog.

### ISSUE THREAD

**Issue by keerthiduvvuri** [microsoft/AzureStorageExplorer#7616]
### Preflight Checklist

- [X] I have installed the [latest version of Storage Explorer](https://github.com/Microsoft/AzureStorageExplorer/releases/latest).
- [X] I have checked existing resources, including the [troubleshooting guide](https://docs.microsoft.com/azure/storage/common/storage-explorer-troubleshooting) and the [release notes](https://github.com/microsoft/AzureStorageExplorer/releases).
- [X] I have searched for [similar issues](https://github.com/Microsoft/AzureStorageExplorer/issues).

### Storage Explorer Version

1.32.1(96)

### Regression From

NO

### Architecture

x64

### Storage Explorer Build Number

20231114.10

### Platform

macOS

### OS Version

MAC Version 13.6.1

### GitHub Tags:
#A11yMAS;#A11yTCS;#MAC;#DesktopApp;#A11ySev2;#BM_AzureToolsStorageExplorer_MAC_Dec2023;#Azure Tools Storage Explorer;#FTP;#Voiceover;#WCAG1.3.1;#Info and Relationship;#Element:Dialog;

### Environment Details:
Application: Microsoft Azure Storage Explore
version: 1.32.1 (96)
Screen Reader: Voice Over

### Steps to Reproduce

1. Launch Storage Explorer
2. Tab Navigate to "Open Connect dialog" pane and Sign into Azure.
3. Tab Navigate to "Manage Accounts" pane and Select the subscriptions you will use. Once selected, click on "Apply" button.
4. Tab Navigate to "Disks" of the selected subscription.
5. Tab Navigate to "A11ytest" and click on it.
6. Tab Navigate to Delete Disk button and click on it.
7. "Delete Disk" Popup gets opened.
8. Turn on Voice Over and Observe that Voice over is not announcing the Name "Microsoft Azure Storage explorer" of the dialog.

### Actual Experience

When User press enter on Delete button a dialog open Voice over is not announcing the Name "Microsoft Azure Storage explorer" of the dialog.

### Expected Experience

Voice over should be announcing the Name "Microsoft Azure Storage explorer" of the dialog.

### User Impact:
Screen reader users will face difficulty if Voice over is not announcing the Name "Microsoft Azure Storage explorer" of the dialog.

### Attachment:
![A11y_Azure Tools Storage Explorer_Disk_Delete Disk_Screen reader](https://github.com/microsoft/AzureStorageExplorer/assets/98728920/fa761d3d-76c7-48a4-93be-87109de2a6a7)

**Comment by craxal** [comment]
@keerthiduvvuri I am not able to reproduce this on Sonoma 14.2.1. VoiceOver announces the title of the dialog as expected. What happens if you switch on VoiceOver before the dialog opens?

**Comment by keerthiduvvuri** [comment]
@craxal as soon as user hit enter on delete button this Microsoft azure storage explorer dialog open and voice over announces as warning one more item, one more item dialog yes button, anywhere it was not announcing the dialog name.

**Comment by craxal** [comment]
@MRayermannMSFT @JasonYeMSFT @richardMSFT Can any of you reproduce this? My Mac is on 14.2.1, so my VoiceOver version may behave differently.

**Comment by JasonYeMSFT** [comment]
Here is my finding when I try to reproduce this issue:

When dialog window shows up, the dialog title is initially set to "Storage Explorer". VoiceOver starts reading it but will be quickly interrupted and starts reading the "warning one more item, one more item dialog yes button" message.

**Comment by craxal** [comment]
This issue is not isolated to React dialogs. It occurs in Knockout dialogs as well. This isn't trivial, as it require making changes to the element with the "dialog" role. The role needs to be configurable, and labels need to be provided.

**Comment by craxal** [comment]
@keerthiduvvuri We have merged a fix for this issue in regard to messages boxes. Please verify and close.

**Comment by keerthiduvvuri** [comment]
##Closed;
